src/CommunityToolkit.Maui/Converters/BaseConverterOneWay.shared.cs

```diff
@@ -18,9 +18,20 @@ public abstract class BaseConverterOneWay<TFrom, TTo> : BaseConverterOneWay
 	/// <param name="parameter">Additional parameter for the converter to handle. This is not implemented.</param>
 	/// <param name="culture">The culture to use in the converter. This is not implemented.</param>
 	/// <returns>An object of type <typeparamref name="TTo"/>.</returns>
-	public override object? Convert(object? value, Type? targetType, object? parameter, CultureInfo? culture)
+	public override sealed object? Convert(object? value, Type? targetType, object? parameter, CultureInfo? culture)
 	{
-		if (value is not TFrom valueFrom)
+		if (value is null && IsNullable<TFrom>())
+		{
+#pragma warning disable CS8604 // Possible null reference argument.
+			return ConvertFrom(default);
+#pragma warning restore CS8604 // Possible null reference argument.
+		}
+		else if (value is null && !IsNullable<TFrom>())
+		{
+			throw new ArgumentException($"value cannot be null because {nameof(TFrom)} is not Nullable");
```

Comment:
Sorry one last thing (I hope) could/should this be `ArgumentNullException` instead here?

Reply:
Good question! I think that's valid. 

[ArgumentNullException](https://docs.microsoft.com/en-us/dotnet/api/system.argumentnullexception?view=net-6.0):

>An [ArgumentNullException](https://docs.microsoft.com/en-us/dotnet/api/system.argumentnullexception?view=net-6.0) exception is thrown when a method is invoked and at least one of the passed arguments is null but should never be null.
An [ArgumentNullException](https://docs.microsoft.com/en-us/dotnet/api/system.argumentnullexception?view=net-6.0) exception is thrown at run time in the following two major circumstances, both of which reflect developer error:
An uninstantiated object is passed to a method. To prevent the error, instantiate the object.
>
> An object returned from a method call is then passed as an argument to a second method, but the value of the original returned object is null. To prevent the error, check for a return value that is null and call the second method only if the return value is not null.
> 
> [ArgumentNullException](https://docs.microsoft.com/en-us/dotnet/api/system.argumentnullexception?view=net-6.0) behaves identically to [ArgumentException](https://docs.microsoft.com/en-us/dotnet/api/system.argumentexception?view=net-6.0). It is provided so that application code can differentiate between exceptions caused by null arguments and exceptions caused by arguments that are not null.

Reply:
Updated ✅